lua/starfall/libs_sv/nextbot.lua

```diff
@@ -0,0 +1,598 @@
+
+local nextbots = {}
+
+local nextbot_library, nb_meta, nb_methods = instance.Libraries.nextbot, instance.Types.NextBot, instance.Types.NextBot.Methods
+local vec_meta, vwrap, vunwrap = instance.Types.Vector, instance.Types.Vector.Wrap, instance.Types.Vector.Unwrap
+local navarea_methods, navarea_meta, navwrap, navunwrap = instance.Types.NavArea.Methods, instance.Types.NavArea, instance.Types.NavArea.Wrap, instance.Types.NavArea.Unwrap
+local nbwrap, nbunwrap = instance.Types.NextBot.Wrap, instance.Types.NextBot.Unwrap
+
+local function nextbotOnDestroy(ent)
+	local ply = instance.player
+	nbCount:free(ply, 1)
+	nextbots[ent] = nil
+end
+
+local function register(ent)
+	ent:CallOnRemove("starfall_nextbot_delete", nextbotOnDestroy)
+	nbCount:free(instance.player, -1)
+	nextbots[ent] = true
+end
+
+instance:AddHook("deinitialize", function()
+	if #nextbots > 0 then
+		for nextbot, _ in pairs(nextbots) do
+			-- For some reason there's numbers being added to the nextbots table when they're created so I have to do this or it errors. Why does it add numbers to the table? I wish I knew.
+			if IsEntity(nextbot) then
+				nextbot:Remove()
+			end
+		end
+	end
+end)
+
+function nb_meta:__tostring()
+	return "NextBot"
+end
+
+--- Creates a customizable NextBot
+-- @server
+-- @param Vector spawnpos The position the nextbot will be spawned at.
+-- @param string? model The model the nextbot will use. If nil, uses Kleiner.
+-- @return NextBot The nextbot.
+function nextbot_library.create(pos, mdl)
+
+	checkpermission(instance, nil, "nextbot.create")
+	checkluatype(mdl, TYPE_STRING)
+	
+	local ply = instance.player
+	nbCount:checkuse(ply, 1)
+
+	local nb = ents.Create("starfall_cnextbot")
+	register(nb, instance)
+	nb:SetPos(vunwrap(pos))
+	nb:SetNBModel(mdl or "models/kleiner.mdl")
+	nb.chip = instance.entity
+	nb:Spawn()
+	table.insert(nextbots, nb)
+	return nbwrap(nb)
+end
+	
+--- Checks if a user can spawn anymore nextbots.
+-- @server
+-- @return boolean True if user can spawn nextbots, False if not.
+function nextbot_library.canSpawn()
+	if not SF.Permissions.hasAccess(instance, nil, "nextbot.create") then return false end
+	return nbCount:check(instance.player) > 0
+end
+
+--- Makes the nextbot try to go to a specified position.
+-- @server
+-- @param Vector gotopos The position the nextbot will continuosly try to go to.
+function nb_methods:setGotoPos(pos)
+		local nb = nbunwrap(self)
+		checkpermission(instance, nb, "nextbot.setGotoPos")
+		nb.goTo = vunwrap(pos)
+end
+
+--- Removes the "go to" position from the NextBot.
+-- @server
+function nb_methods:removeGotoPos()
+	local nb = nbunwrap(self)
+	checkpermission(instance, nb, "nextbot.removeGotoPos")
+	nb.goTo = nil
+end
+
+--- Returns the Vector the nextbot is trying to go to, set by setGotoPos
+-- @server
+-- @return Vector Vector nextbot is trying to go to if it exists, else returns nil.
+function nb_methods:isGoingTo()
```

Comment:
`-- @return Vector? Where the nextbot is trying to go to if it exists, else returns nil.`

Also change the function name to getGoingTo